components/board/numato/board.go

```diff
@@ -354,12 +356,44 @@ type analog struct {
 	pin string
 }
 
-func (a *analog) Read(ctx context.Context, extra map[string]interface{}) (int, error) {
+// analog Read returns the analog value with the range and step size in mV/bit.
+func (a *analog) Read(ctx context.Context, extra map[string]interface{}) (int, board.AnalogRange, error) {
 	res, err := a.b.doSendReceive(ctx, fmt.Sprintf("adc read %s", a.pin))
 	if err != nil {
-		return 0, err
+		return 0, board.AnalogRange{}, err
+	}
+	reading, err := strconv.Atoi(res)
+	if err != nil {
+		return 0, board.AnalogRange{}, err
 	}
-	return strconv.Atoi(res)
+	var max float32
+	var stepSize float32
+	switch a.b.productID {
+	case 0x805:
+		// 128 channel usb numato has 12 bit resolution
+		max = 3.3
+		stepSize = max / 4096
+	case 0x802:
+		// 32 channel usb numato has 10 bit resolution
+		max = 3.3
+		stepSize = max / 1024
+	case 0x800:
+		// 8 and 16 pin usb versions have the same product ID but different voltage ranges
+		// both have 10 bit resolution
+		if a.b.pins == 8 {
+			max = 5.0
+		} else if a.b.pins == 16 {
+			max = 3.3
+		}
+		stepSize = max / 1024
+	case 0xC05:
+		// 1 channel usb relay module numato - 10 bit resolution
+		max = 5.0
+		stepSize = max / 1024
+	default:
+	}
+	stepSize *= 1000
```

Comment:
Why are we multiplying `stepSize` by a thousand? The whole idea of "maximum voltage divided by maximum bit size" seemed like a good idea to me.

Reply:
the scope doc converted the example calculation to `mV/bit`, I think it makes it more readable otherwise its very small values but willing to discuss changing.

Reply:
That makes sense! Put a comment here pointing out the right units.

(apologies for the confusion where I remarked on this in the test)

Reply:
...wait, so Min and Max are in volts, but StepSize is in millivolts? Put a comment in the definition of the struct explaining that.

...or rather, it seems surprising that they have different units, and perhaps the way to make it less surprising, instead of documenting the surprise, is to change the units so they're either all volts or all millivolts. I checked the scope doc, but it doesn't explicitly mention the units on any of these values (two examples mention mV and rejected alternative 3 mentions units, but I don't think this was ever laid out clearly).

Reply:
yeah thats fair ill just keep it in volts/bit